pkg/util/mount/mount_windows_test.go

```diff
@@ -601,3 +601,42 @@ func TestGetFileType(t *testing.T) {
 		}
 	}
 }
+
+func TestFormatAndMount(t *testing.T) {
+	fakeMounter := ErrorMounter{&FakeMounter{}, 0, nil}
+	fakeExec := NewFakeExec(nil)
+	mounter := SafeFormatAndMount{
+		Interface: &fakeMounter,
+		Exec:      fakeExec,
+	}
+
+	tests := []struct {
+		device       string
+		target       string
+		fstype       string
+		mountOptions []string
+		expectError  bool
+	}{
+		{
+			"0",
+			`c:\var\lib\kubelet\pods\146f8428-83e7-11e7-8dd4-000d3a31dac4\volumes\kubernetes.io~azure-disk`,
+			"NTFS",
+			[]string{},
+			false,
+		},
+		{
+			"invalidDevice",
+			`c:\var\lib\kubelet\pods\146f8428-83e7-11e7-8dd4-xxxd3a31dac4\volumes\kubernetes.io~azure-disk`,
+			"NTFS",
+			[]string{},
+			true,
+		},
+	}
+
+	for _, test := range tests {
+		err := mounter.FormatAndMount(test.device, test.target, test.fstype, test.mountOptions)
```

Comment:
do these unit tests actually format devices?

Reply:
no, it won't format device, actually it's using `fakeExec`, need an e2e windows test for this kind of real case. I know cloudbase is driving on e2e windows test now, may catch up on that later.

Reply:
Is it possible to add unit tests for conditions where formatting is needed vs not?

Reply:
@msau42 thanks for the check, although the format won't happen, while I made a few changes on the unit test to let other operations happen, and finally `FormatAndMount` will complete successfully. This test case depends on C driver in which in normal condition, C driver will always exists on a windows machine.

Reply:
What if you used FakeExec, and then just validated the command is correct?

Reply:
@msau42 good point, I have added a FakeExec in test, PTAL